src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```diff
@@ -380,32 +359,38 @@ private PluginDescriptionFile findPluginDescription(Class<? extends JavaPlugin>
 	public void callEvent(Event event) throws IllegalStateException
 	{
 		events.add(event);
-		for (List<ListenerEntry> listeners : eventListeners.values())
-		{
-			for (ListenerEntry entry : listeners)
-			{
-				if (entry.isCompatibleFor(event))
-				{
-					entry.invokeUnsafe(event);
-				}
+		HandlerList handlers = event.getHandlers();
+		RegisteredListener[] listeners = handlers.getRegisteredListeners();
+		Arrays.stream(listeners).forEach(l ->callRegisteredListener(l,event));
```

Comment:
Try to avoid Streams and inline `.forEach` whenever you can.
They are significantly slower and much less optimized by the JVM than a standard for-loop.
We are talking micro-optimizations here but since the Bukkit API is heavily centered around the Event contract, this method will be a hot point of contact and should be handled with care.
So if you wanna reduce the load of this method, you should consider refactoring it back to a normal for-loop.